src/main/java/com/gradle/enterprise/conventions/customvalueprovider/GradleEnterpriseConventions.java

```diff
@@ -81,25 +83,21 @@ public boolean isCiServer() {
     }
 
     public String getEnvVariableThenSystemProperty(String envName, String systemPropertyName, String defaultValue) {
-        return getProviderFactory().environmentVariable(envName).forUseAtConfigurationTime()
-            .orElse(getProviderFactory().systemProperty(systemPropertyName).forUseAtConfigurationTime())
-            .orElse(defaultValue).get();
+        String value = getEnv(envName);
```

Comment:
🤔 Would it make sense to continue using the `Provider`s here? Like this, all these things will be configuration inputs. Not sure if we can avoid that.

Reply:
>Would it make sense to continue using the Providers here?

If you obtain the provider at configuration time, it will be an input anyway. Our current stance is to prefer Java APIs unless all you do is connecting the external source to task input - which isn't the case here.

Reply:
Alright, let's leave it as is. I'd probably move it to `buildScan.background()`, so it isn't tracked as an input anymore. We can do that in a follow up PR when we have the first round of changes in and see that everything still works.